contracts/src/v0.8/ccip/pools/GHO/UpgradeableBurnMintTokenPool.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import {ITypeAndVersion} from "../../../shared/interfaces/ITypeAndVersion.sol";
+import {IBurnMintERC20} from "../../../shared/token/ERC20/IBurnMintERC20.sol";
+
+import {UpgradeableBurnMintTokenPoolAbstract} from "./UpgradeableBurnMintTokenPoolAbstract.sol";
+import {UpgradeableTokenPool} from "./UpgradeableTokenPool.sol";
+
+import {Initializable} from "solidity-utils/contracts/transparent-proxy/Initializable.sol";
+import {IRouter} from "../../interfaces/IRouter.sol";
+
+/// @title UpgradeableBurnMintTokenPool
+/// @author Aave Labs
+/// @notice Upgradeable version of Chainlink's CCIP BurnMintTokenPool
+/// @dev Contract adaptations:
+/// - Implementation of Initializable to allow upgrades
+/// - Move of allowlist and router definition to initialization stage
+/// - Add GHO-Specific onlyOwner `directMint` which mints liquidity to the old pool &
+/// increases facilitator level.
+/// - Add GHO-Specific onlyOwner `directBurn` which burns liquidity & reduces facilitator level.
+/// - Remove i_token decimal check in UpgradeableTokenPool constructor
```

Comment:
do we wanna do like in https://github.com/aave/ccip/pull/21/files#diff-20977dcbbf1840415b3d5f5a44b617518a16b37b64fd342fec75b2b5faeedec5R21 and point to the inherited contract?

Reply:
[4e0e90f](https://github.com/aave/ccip/pull/20/commits/4e0e90fbded84e592affeea4dacf5e597f6a251b)